### ISSUE TITLE
R10k can’t be installed with Ruby 3.x

### ISSUE THREAD

**Issue by akerl** [adrienthebo/r10k#1294]
## Describe the Bug

I’m trying to run r10k on Archlinux, which ships with Ruby 3.x. Installing r10k fails as follows:

```
Bundler found conflicting requirements for the Ruby version:
  In Gemfile:
    r10k (~> 3.14.2) was resolved to 3.14.2, which depends on
      cri (= 2.15.10) was resolved to 2.15.10, which depends on
        Ruby (~> 2.3)

  Current Ruby version:
    Ruby (= 3.0.4)
```

## Expected Behavior

When on newer Ruby, r10k uses a newer cri version so that it can be installed.

## Steps to Reproduce

1. On a system with Ruby 3.x, create a Gemfile with “r10k” in it, attempt to bundle install2. 
## Environment
 - Version: 3.14.2
 - Platform: Archlinux

**Comment by mwaggett** [comment]
Hi @akerl ,

Unfortunately, we need to stay pinned to cri 2.15.10 in order to support older Ruby versions (2.3 and 2.4), which we don't feel comfortable dropping support for until we release r10k 4.0. 

Fortunately, we _do_ have a `4.x` branch of this repo with our planned changes for the 4.0 release, including Ruby 3 support. You should be able to build your own gem from that branch, or run from source. 

I hope that helps!

**Comment by akerl** [comment]
It seems like the other viable option is to have the gemspec conditionally pick a newer version of cri if RUBY_VERSION >= 3.0.0?

**Comment by mwaggett** [comment]
@akerl We tried that, but discovered that gemspecs are evaluated at build time, so it will look at the ruby version of the system building the gem, rather than the one installing it: https://github.com/puppetlabs/r10k/commit/8c0821d8b152d658e6178f80c53e86c3fc6c40de

**Comment by bastelfreak** [comment]
@akerl because of that I maintain the r10k package on Arch Linux :) 
https://archlinux.org/packages/community/any/r10k/

**Comment by mwaggett** [comment]
This has been resolved as of r10k 3.15.0.
